deltacat/utils/pyarrow.py

```diff
@@ -33,15 +33,74 @@
 from deltacat.utils.common import ContentTypeKwargsProvider, ReadKwargsProvider
 from deltacat.utils.performance import timed_invocation
 from deltacat.utils.daft import daft_s3_file_to_table
-from deltacat.utils.arguments import sanitize_kwargs_to_callable
+from deltacat.utils.arguments import (
+    sanitize_kwargs_to_callable,
+    sanitize_kwargs_by_supported_kwargs,
+)
 
 logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
 
+
+def _filter_schema_for_columns(schema: pa.Schema, columns: List[str]) -> pa.Schema:
+
+    target_schema_fields = []
+
+    for column_name in columns:
+        index = schema.get_field_index(column_name)
+
+        if index != -1:
+            target_field = schema.field(index)
+            target_schema_fields.append(target_field)
+
+    target_schema = pa.schema(target_schema_fields, metadata=schema.metadata)
+
+    return target_schema
```

Comment:
So what exactly is this function doing? It might be good to add a docstring. Why would we want to remove columns from the schema?

Reply:
Creates a new schema that adheres to the given column name ordering and types.